File: main/fibonacci-modified/fibonacci-modified.py
#!/usr/bin/env python3


def read_record():
    """Reads a line of input as a sequence of integers."""
    return map(int, input().split())


def fib_modified(a, b, n):
    """Computes the modified Fiboniacci-style function."""
    for _ in range(2, n):
        a, b = b, a + b * b
    return b


if __name__ == '__main__':
    print(fib_modified(*read_record()))

File: main/fibonacci-modified/fibonacci_modified.rb
#!/usr/bin/env ruby

def fib_modified(a, b, n)
    (n - 2).times { a, b = b, a + b**2 }
    b
end

def read_record()
    gets.split.map(&:to_i)
end

puts fib_modified(*read_record)
